### ISSUE TITLE
New feature: use `postcss-html` to parse HTML for CLI

### ISSUE THREAD

**Issue by gucong3000** [stylelint/stylelint#2589]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

- New feature for CLI, support `*.vue` & HTML file.
- `stylelint-processor-html` Incompatible with autofixes, and don't match `lang` attribute of  `<style>` tag.

I'd like to push an PR for this.

> What CSS is needed to reproduce this issue?

CSS in `*.vue` file

```html
<style lang="scss">
//sass style
//-----------------------------------
$fontStack:    Helvetica, sans-serif;
$primaryColor: #333;

body {
  font-family: $fontStack;
  color: $primaryColor;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

any

> Which version of stylelint are you using?

any

> How are you running stylelint: CLI, PostCSS plugin, Node API?

stylelint: CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, any non-standard syntax in `*.vue` file

> What did you expect to happen?

e.g. "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
stylelint test/fixtures/vue/test.vue
test/fixtures/vue/test.vue
 14:1  ×  Unknown word   CssSyntaxError
```

**Comment by evilebottnawi** [comment]
@gucong3000 we don't use `processors` in core, this complicates the maintenance of the project, it is better to do it in a separate repository.

**Comment by jeddy3** [comment]
@gucong3000 Thanks for the suggestion.

> - New feature for CLI, support `*.vue` & HTML file.
> - `stylelint-processor-html` Incompatible with autofixes, and don't match `lang` attribute of  `<style>` tag.

Feel free to [create a new processor](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/processors.md) with these capabilities. Perhaps call it `stylelint-processor-style-tags`. Also, feel free to create a PR that adds the new processor to [the list in the docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/processors.md).
